plugins/main/ptp/ptp.go

```diff
@@ -136,13 +137,6 @@ func setupContainerVeth(netns ns.NetNS, ifName string, mtu int, pr *current.Resu
 			}
 		}
 
-		// Send a gratuitous arp for all v4 addresses
```

Comment:
We needed this for the case when an IP is reused very quickly, so the arp tables are stale. I'm not sure we can get rid of it.

Reply:
Based on the route configuration in container namespace, we only need to resolve MAC address for gateway (with .1 address). I'm not sure why sending gratuitous ARP on behalf of container IP could help when IP is reused.

I actually ran the test with and without gratuitous ARP and both passed. Please let me know you want me to revert the change on gratuitous ARP. Thanks!

Reply:
The gratuitous arp is to prevent a timing bug with arp caching - it's not necessary for reachability in the normal case. This was an issue reported long-ago in containernetworking/cni#88 and containernetworking/cni#194.

So, yes, it will work without it, but it does help solve that bug in degenerate cases. And sorry for not writing a better comment :-)

Reply:
Correct - we send the gratuitous arp from the container to the host.

Reply:
I also updated the comments accordingly. Please let me know any further comment. Thanks!